### DISCUSSION TITLE
New in v8.6.0 - ✨Improved Agentic Workflows

### DISCUSSION THREAD

**Discussion by olimorris** [olimorris/codecompanion.nvim#333] [Announcements]
I've been passionate about agentic workflows for some time. Giving the LLM the ability to self-reflect leads to some interesting [results](https://www.deeplearning.ai/the-batch/issue-242/).

In this release I've overhauled how the plugin handles agentic workflows...infact you can create them yourself by adding them to the prompt library (check out the RECIPES and the config.lua file)

https://github.com/user-attachments/assets/c0af5487-221b-4ecf-a604-a2f86dc86c9b

In this video, you'll notice that my next prompts are auto-populated (they could also be auto-submitted but I like to press that `<return>` key myself). The wording of these prompts also give the LLM the opportunity to self-reflect and revise its answer.

I'm keen to implement some of the features that [Aider](https://aider.chat) has in the coming weeks and I think workflows will be a key part of this.

Please play around with workflows and share yours with the wider community.

**Comment by e2r2fx** [comment]
@olimorris see you're playing with the @img tool thing. I can submit that as a PR np, the downside is that its hardcoded to work with OpenAI's vision API with a curl command, but other than that I find it really useful on my end

**Reply by olimorris** [reply]
Ahhh I was just testing custom tooling. I would love to see your img implementation though

**Reply by e2r2fx** [reply]
sure, here's a [gist](https://gist.github.com/e2r2fx/72116fdc3560912b70ee9457e30cf8f4)

The way I use it is quite rudimentary, I just tell the llm to use the @img tool and do X with it, and it builds a query for the vision API.

For providing an image from a screenshot I use a quick n dirty plugin I wrote some time ago
https://github.com/e2r2fx/cloudy.nvim

**Reply by e2r2fx** [reply]
I like better the approach of locally base64 encoding the image and sending that to the vision API. But since I had that plugin laying around... it gets the job done :)
